src/components/char-check-limit/_macro.spec.js

```diff
@@ -21,40 +21,70 @@ describe('FOR: Macro: CharCheckLimit', () => {
             });
             test('THEN: has the data attribute which defines charCountPlural', () => {
-                expect($('.ons-input__limit').attr('data-charcount-plural')).toBe('You have {x} characters remaining');
+                expect($('.ons-input__limit').attr('data-message-plural')).toBe('You have {x} characters remaining');
             });
             test('THEN: has the data attribute which defines charCountSingular', () => {
-                expect($('.ons-input__limit').attr('data-charcount-singular')).toBe('You have {x} character remaining');
+                expect($('.ons-input__limit').attr('data-message-singular')).toBe('You have {x} character remaining');
             });
             test('THEN: has the data attribute which defines charCountOverLimitSingular', () => {
-                expect($('.ons-input__limit').attr('data-charcount-limit-singular')).toBe('{x} character too many');
+                expect($('.ons-input__limit').attr('data-message-over-limit-singular')).toBe('{x} character too many');
             });
             test('THEN: has the data attribute which defines charCountOverLimitPlural', () => {
-                expect($('.ons-input__limit').attr('data-charcount-limit-plural')).toBe('{x} characters too many');
+                expect($('.ons-input__limit').attr('data-message-over-limit-plural')).toBe('{x} characters too many');
+            });
+            test('THEN: sets char as the count type to count number of characters', () => {
+                expect($('.ons-input__limit').attr('data-count-type')).toBe('char');
             });
         });
     });
 
     describe('GIVEN: Params: variant', () => {
-        describe('WHEN: variant is provided', () => {
-            const $ = cheerio.load(
-                renderComponent(
-                    'char-check-limit',
-                    {
-                        ...EXAMPLE_CHAR_CHECK_LIMIT,
-                        variant: 'check',
-                    },
-                    ['<p>Test content.</p>'],
-                ),
-            );
-
-            test('THEN: passes jest-axe checks with variant set to check', async () => {
+        describe('WHEN: variant is set to word', () => {
+            const $ = cheerio.load(renderComponent('char-check-limit', EXAMPLE_WORD_LIMIT));
+
+            test('THEN: passes jest-axe checks', async () => {
                 const results = await axe($.html());
                 expect(results).toHaveNoViolations();
             });
 
-            test('THEN: renders the passed content', () => {
-                expect($('.ons-js-char-check-input').text()).toBe('Test content.');
+            test('THEN: sets word as the count type to count number of words', () => {
+                expect($('.ons-input__limit').attr('data-count-type')).toBe('word');
+            });
+        });
+    });
+
+    describe('GIVEN: Params: wordCountPlural', () => {
```

Comment:
These tests could be restructured to have less repetition they could all come under the same section as variant and then you would only need line 43 once